internal/go_repository.bzl

```diff
@@ -258,7 +247,6 @@ go_repository = repository_rule(
             ],
         ),
         "build_extra_args": attr.string_list(),
-        "build_config": attr.label(default = "@//:WORKSPACE"),
```

Comment:
Leave this attribute in place, but change the default to `@bazel_gazelle_go_repository_config//:WORKSPACE`.

We should avoid making unnecessary breaking changes. People may be using this for custom per-rule configuration.

Also, this is important for efficiency: repository rules are re-evaluated when any file referenced by a label attribute or `ctx.path` with a dynamic label is changed. In the latter case, the whole repository rule is re-evaluated, and any long-running operations (like the `fetch_repo` download) are re-done. So if you ever need to do `ctx.path(Label("..."))`, it should be at the beginning of the function, before anything expensive.